hircluster.c

```diff
@@ -3773,6 +3697,124 @@ static void redisClusterAsyncCallback(redisAsyncContext *ac, void *r,
     cluster_async_data_free(cad);
 }
 
+/* Reply callback function for CLUSTER SLOTS */
+void clusterSlotsReplyCallback(redisAsyncContext *ac, void *r, void *privdata) {
+    UNUSED(ac);
+    redisReply *reply = (redisReply *)r;
+    redisClusterAsyncContext *acc = (redisClusterAsyncContext *)privdata;
+    redisClusterContext *cc = acc->cc;
+
+    if (reply == NULL) {
+        /* Retry using available nodes */
+        if (updateSlotMapAsync(acc) == REDIS_OK) {
+            acc->update_route_time = SLOTMAP_UPDATE_ONGOING;
+        } else {
+            acc->update_route_time =
+                hi_usec_now() + SLOTMAP_UPDATE_THROTTLE_USEC;
+        }
+        return;
+    }
+
+    dict *nodes = parse_cluster_slots(cc, reply, cc->flags);
+    if (updateNodesAndSlotmap(cc, nodes) != REDIS_OK) {
+        /* Ignore failures for now */
+    }
+    acc->update_route_time = hi_usec_now() + SLOTMAP_UPDATE_THROTTLE_USEC;
+}
+
+/* Reply callback function for CLUSTER NODES */
+void clusterNodesReplyCallback(redisAsyncContext *ac, void *r, void *privdata) {
+    UNUSED(ac);
+    redisReply *reply = (redisReply *)r;
+    redisClusterAsyncContext *acc = (redisClusterAsyncContext *)privdata;
+    redisClusterContext *cc = acc->cc;
+
+    if (reply == NULL) {
+        /* Retry using available nodes */
+        if (updateSlotMapAsync(acc) == REDIS_OK) {
+            acc->update_route_time = SLOTMAP_UPDATE_ONGOING;
+        } else {
+            acc->update_route_time =
+                hi_usec_now() + SLOTMAP_UPDATE_THROTTLE_USEC;
+        }
+        return;
+    }
+
+    dict *nodes = parse_cluster_nodes(cc, reply->str, reply->len, cc->flags);
+    if (updateNodesAndSlotmap(cc, nodes) != REDIS_OK) {
+        /* Ignore failures for now */
+    }
+    acc->update_route_time = hi_usec_now() + SLOTMAP_UPDATE_THROTTLE_USEC;
```

Comment:
This field, does it mean "when can we update slots next time"?

I think it's better that we store the current time, i.e. what time the slots were updated. This better matches lastConnectionAttempt we have on a node.

Reply:
Updated to use a lastSlotmapUpdateAttempt set with current time